src/routing_node.rs

```diff
@@ -167,18 +167,19 @@ impl RoutingNode {
     fn handle_new_connection(&mut self, endpoint : Endpoint) {
         debug!("New connection on {:?}", endpoint);
         // only accept new connections if we are a full node
-        let has_bootstrap_endpoints = self.core.has_bootstrap_endpoints();
-        if !self.core.is_node() {
-            if has_bootstrap_endpoints {
-                // we are bootstrapping, refuse all normal connections
-                self.connection_manager.drop_node(endpoint);
-                return;
-            } else {
-                let assigned_name = NameType::new(crypto::hash::sha512::hash(
-                    &self.core.id().name().0).0);
-                let _ = self.core.assign_name(&assigned_name);
-            }
-        }
+        // FIXME(dirvine) I am not sure we should not accept connections here :16/08/2015
+        // let has_bootstrap_endpoints = self.core.has_bootstrap_endpoints();
+        // if !self.core.is_node() {
+        //     if has_bootstrap_endpoints {
+        //         // we are bootstrapping, refuse all normal connections
+        //         self.connection_manager.drop_node(endpoint);
+        //         return;
+        //     } else {
```

Comment:
We can debate whether "normal connections" should not be accepted, but by disabling the self_assignment of a name the network will for sure not start, as everyone will stay a Client, and no-one will become a Node.

The condition upon which to refuse connections here is very subtle, and tests out to be correct - at least in the current implementation:
 if ! core.is_node()  ===>   if we do not have a Node(name)/ a routing table
    if has_bootstrap_endpoints ===> but we do have connections that are labelled as bootstrap;
===> we are connected to relay nodes (either trying to request a network name, or as pure clients), so we should not attempt to help other nodes bootstrap into the network, as we are ourselves still clients